### DISCUSSION TITLE
Launchers vs. Samples

### DISCUSSION THREAD

**Discussion by ndr-brt** [eclipse-edc/Connector#196] [General]
What's the reason to keep two different folders for these modules?
Is `launchers` folder still useful? Aren't launchers samples themselves?
e.g.
`basic` launcher is the same as the sample `01-basic-connector` but with some added dependencies
`ids-connector` launcher is also a sample
`junit` launcher is not really a launcher, it contains the `EdcExtension` class that is an util class and maybe it could be moved into a "test-util" lib
`registration-service-app` launcher is also a sample.

So I think we could move all the launchers into the sample folder

**Comment by jimmarino** [comment]
I'll let @paullatzelsperger comment on the launcher structure as he had a proposal for simplifying that. 

Regarding JUnit, I do think EdcExtension fits the bill of a launcher because it bootstraps a JUnit runtime. It shouldn't be in a lib module since those types of modules should only contain utility code/functions, not runtime-related code.

**Comment by paullatzelsperger** [comment]
@ndr-brt so they're not the same :)
- The numbered samples are accompanied by documentation, and they are also incremental.
- The `other` samples focus on very specific use cases, typically containing some code. 
- the `launchers` directory contains build files for various EDC runtimes that can be used as starting points by anyone who wants to do so. They are mere templates.

Launchers are essentially build files, with the option to add custom startup code (the `*Runtime.java` classes).

We can move the launchers to the samples directory for all I care so long as we remain aware of the fact that there is a difference between them and the other samples.

**Reply by ndr-brt** [reply]
Why not move the `onboarding` samples into a subfolder and avoid to have an `other` folder that explains nothing about its content, moving all the samples contained into it to the `sample` folder?
e.g.:
```
samples
  onboarding
    01-basic-connector
    02-health-endpoint
    ...
  commandline
  copy-between-azure-and-s3
  ...
```
